packages/core/src/sync-realtime/index.ts

```diff
@@ -833,28 +910,16 @@ export const createRealtimeSync = (
         } catch (_error) {
           if (isKilled) return;
 
-          const error = _error as Error;
-
-          args.common.logger.warn({
-            service: "realtime",
-            msg: `Failed to fetch latest '${args.network.name}' block`,
-            error,
-          });
-
-          // After a certain number of attempts, emit a fatal error.
-          if (++consecutiveErrors === ERROR_TIMEOUT.length) {
-            args.common.logger.error({
-              service: "realtime",
-              msg: `Fatal error: Unable to fetch latest '${args.network.name}' block after ${ERROR_TIMEOUT.length} attempts.`,
-              error,
-            });
-
-            args.onFatalError(error);
-          }
+          onError(_error as Error);
         }
       };
 
-      interval = setInterval(enqueue, args.network.pollingInterval);
+      if (isPolling) {
+        interval = setInterval(enqueue, args.network.pollingInterval);
+      } else {
+        // Check on the connection every second
+        interval = setInterval(watchWebsocket, 1000);
```

Comment:
Do we need to do this? Looking at the viem implementation of `watchBlocks` seems like they are handling it differently

https://github.com/wevm/viem/blob/main/src/actions/public/watchBlocks.ts

Reply:
My idea behind this implementation was to have persistency of websocket connection, so if error happens during active connection, we drop it and reconnect again. From the watchblocks implementation, they seem to leave this flexibility via onError callback.